File: CrackingTheCodingInterview/Udemy/LevelOrderTraversal.java
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

import javax.swing.tree.TreeNode;

public class LevelOrderTraversal {
    public List<List<Integer>> levelOrder(TreeNode root) {
        if (root == null) {
            return new ArrayList<List<Integer>>();
        }
        List<List<Integer>> res = new ArrayList<>();
        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);
        while (queue.size() != 0) {
            List<Integer> currentLevelValues = new ArrayList<>();
            int count = 0;
            int length = queue.size();
            while (count < length) {
                TreeNode currentNode = queue.remove();
                currentLevelValues.add(currentNode.val);
                if (currentNode.left != null) {
                    queue.add(currentNode.left);
                }
                if (currentNode.right != null) {
                    queue.add(currentNode.right);
                }
                count++;
            }
            res.add(currentLevelValues);
        }
        return res;
    }
}

File: CrackingTheCodingInterview/LevelOrderTraversal.js
const levelOrder = function(root) {
    if (!root) {
        return [];
    }
    let res = [];
    const queue = [root];
    while (queue.length) {
        let length = queue.length;
        let count = 0;
        const currentLevelValues = [];
        while (count < length) {
            const currentNode = queue.shift();
            currentLevelValues.push(currentNode.value);
        }
    }
}